substrate/frame/alliance/src/tests.rs

```diff
@@ -258,51 +248,46 @@ fn close_works() {
 			proposal_len
 		));
 
-		let record = |event| EventRecord { phase: Phase::Initialization, event, topics: vec![] };
-		assert_eq!(
-			System::events(),
-			vec![
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Proposed {
-					account: 1,
-					proposal_index: 0,
-					proposal_hash: hash,
-					threshold: 3
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
-					account: 1,
-					proposal_hash: hash,
-					voted: true,
-					yes: 1,
-					no: 0,
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
-					account: 2,
-					proposal_hash: hash,
-					voted: true,
-					yes: 2,
-					no: 0,
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
-					account: 3,
-					proposal_hash: hash,
-					voted: true,
-					yes: 3,
-					no: 0,
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Closed {
-					proposal_hash: hash,
-					yes: 3,
-					no: 0,
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Approved {
-					proposal_hash: hash
-				})),
-				record(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Executed {
-					proposal_hash: hash,
-					result: Ok(()),
-				}))
-			]
-		);
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(
+			AllianceMotionEvent::Proposed {
+				account: 1,
+				proposal_index: 0,
+				proposal_hash: hash,
+				threshold: 3,
+			},
+		));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
+			account: 1,
+			proposal_hash: hash,
+			voted: true,
+			yes: 1,
+			no: 0,
+		}));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
+			account: 2,
+			proposal_hash: hash,
+			voted: true,
+			yes: 2,
+			no: 0,
+		}));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Voted {
+			account: 3,
+			proposal_hash: hash,
+			voted: true,
+			yes: 3,
+			no: 0,
+		}));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(AllianceMotionEvent::Closed {
+			proposal_hash: hash,
+			yes: 3,
+			no: 0,
+		}));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(
+			AllianceMotionEvent::Approved { proposal_hash: hash },
+		));
+		System::assert_has_event(mock::RuntimeEvent::AllianceMotion(
+			AllianceMotionEvent::Executed { proposal_hash: hash, result: Ok(()) },
+		));
```

Comment:
Does the order of these matter? Just wondering why you switched from the old syntax (where order matters) to this one one where it seemingly doesn't.

Reply:
I don't like this either, but I couldn't find an alternative. 
The previous syntax was matching the whole events list, and the alternative was to add Preimage's events to this event chain, which didn't seem that useful to me